### PR TITLE
Null facet

### PR DESCRIPTION
**Pull request by RFSH**
1. Hide them based on the two newly added ermrestjs APIs.
2. Show null on top of the other options.
3. Show null option in the modal beside the not-null

I just need to add one extra test case for testing the "null" option in the modal.

#1640

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -82,6 +82,7 @@ before_install:
   - cd ..
   - git clone https://github.com/informatics-isi-edu/ermrestjs.git
   - cd ermrestjs
+  - git checkout performance-fix
```
Comment (jrchudy) [inline .travis.yml]: Don't forget to change this back.

File: common/faceting.js [hunk 2]
```diff
@@ -851,12 +855,16 @@
                     defer.resolve();
                 }
                 else {
+                    if (scope.facetColumn.hasNullFilter) {
+                        scope.facetModel.appliedFilters.push(facetingUtils.getNullFilter(true));
```
Comment (jrchudy) [inline common/faceting.js]: Does this mean it should be selected by default when the facetColumn has it? I thought if it was defined for the facetColumn that just means it should be shown..

File: common/faceting.js [hunk 3]
```diff
@@ -924,8 +945,15 @@
 
 
                 var appliedLen = scope.facetModel.appliedFilters.length;
+                var pageSize = tableConstants.PAGE_SIZE;
+                if (!scope.facetColumn.hideNullChoice) {
```
Comment (jrchudy) [inline common/faceting.js]: Could you add a comment as to why the page size should be increased because of this filter. I was thinking the presence of this filter shouldn't change the page size used in the read function..

File: common/modal.js [hunk 4]
```diff
@@ -215,12 +216,11 @@
...
         function getMultiSelectionResult() {
-            var res = vm.tableModel.selectedRows;
-            if (!Array.isArray(res)) res = [];
             if (vm.tableModel.matchNotNull) {
-                res = {matchNotNull: true};
+                return {matchNotNull: true};
             }
-            return res;
+            var res = vm.tableModel.selectedRows;
+            return Array.isArray(res) ? res : [];
```
Comment (jrchudy) [inline common/modal.js]: Returning an array here, but above you return an object..

File: common/templates/recordsetSelectFaceting.html [hunk 5]
```diff
@@ -17,8 +17,8 @@
                         <a class="remove-link" ng-click="removePill(row.uniqueId, $event)"><i class="glyphicon glyphicon-remove"></i></a>
                         <span ng-if="row.displayname.value && row.displayname.isHTML" ng-bind-html="row.displayname.value"></span>
                         <span ng-if="row.displayname.value && !row.displayname.isHTML" ng-bind="row.displayname.value"></span>
-                        <span ng-if="row.displayname.value == '' && row.uniqueId != null" ng-bind-html="defaultDisplayname.empty"></span>
-                        <span ng-if="row.displayname.value == null || row.uniqueId === null" ng-bind-html="defaultDisplayname.null"></span>
+                        <span ng-if="row.displayname.value == ''" ng-bind-html="defaultDisplayname.empty"></span>
```
Comment (jrchudy) [inline common/templates/recordsetSelectFaceting.html]: why don't we need to check uniqueId anymore?

File: common/faceting.js (hunk 3 above)
Comment (RFSH) [reply common/faceting.js]: This should be refactored.

File: common/modal.js (hunk 4 above)
Comment (RFSH) [reply common/modal.js]: This can be refactored.
